### ISSUE TITLE
Add `update` or `map` function to ease bounded number manipulation

### ISSUE THREAD

**Issue by hugooliveirad** [jschomay/elm-bounded-number#1]
Hey! First of all, great work on `elm-bounded-number` using it, loving it.

It is very common for having a way to map a value of a data structure. Most data structures allow it, such as `Maybe`, `Result`, `Dict`, and so on. It would be great to have an `update` or `map` function so that you can more easily chain bounded number operations.

I have added a function similar to this one to one of the projects I'm working on:

```elm
mapBounded fn bounded =
    let
        value = Number.Bounded.value bounded
    in
    Number.Bounded.set (fn value) bounded
```

This let's me refactor things like this:

```diff
-operation : Int -> Int
+operation : Bounded Int -> Int
 operation =
-    (*) 20 >> (+) 10
+    Number.Bounded.update <| (*) 20 >> (+) 10
```

It also makes it easier to refactor from `Number.Bounded.inc`. Such as: https://github.com/hugobessaa/nanotimers/pull/11/commits/696171908045a3d369a25dd0855ec8eccf474e0e

I'm not sure if the proper name should be either `map` or `update`.

What do you think about adding this function? I could send a PR.

**Comment by jschomay** [comment]
Sorry for the delay, I thought I had responded already!

Hi Hugo. That is a good suggestion. If you make a pr with the code you
suggested I will merge it. Call it `map`. And be sure to make a test for it
too.

Thanks! I'm glad you are finding it useful. I wasn't sure if anyone used
this library! (out of curiosity, what is your use case?)

I'll look for your pr :)

**Comment by hugooliveirad** [comment]
Never feel sorry for delayed replies on open source work 😄 

My use case is that I have timers and the amount of seconds left in the timer should never be below 0.

I'll work on a PR and send it your way soon.

**Comment by jschomay** [comment]
Published to 2.1.0
